src/Microsoft.OData.Core/UriParser/SemanticAst/SelectExpandClauseExtensions.cs

```diff
@@ -106,12 +106,10 @@ internal static void Traverse<T>(this SelectExpandClause selectExpandClause, Fun
             {
                 string currentExpandClause = String.Join("/", expandSelectItem.PathToNavigationProperty.WalkWith(PathSegmentToStringTranslator.Instance).ToArray());
                 T subResult = default(T);

```

Comment:
> = default(T) [](start = 27, length = 13)

Since you are no longer conditionally calling Traverse, do you still need to initialize subResult?